lib/widgets/poll.dart

```diff
@@ -44,21 +48,34 @@ class _PollWidgetState extends State<PollWidget> {
     });
   }
 
+  void _toggleVote(PollOption option) async {
+    final store = PerAccountStoreWidget.of(context);
+    // The poll data in store might be obselete before we get the event
+    // that updates it.  This is fine because the result will be consistent
+    // eventually, regardless of the possible duplicate requests.
+    final op = widget.poll.hasUserVotedFor(userId: store.selfUserId, key: option.key)
+      ? PollVoteOp.remove
```

Comment:
I don't quite follow what this comment is saying. What's the scenario where it's obsolete, and when?

If this is intended as reasoning for why we're determining `op` based on the data in the store, rather than something else like telling the server to toggle from whatever state the server has:

I think the real point there is that when the UI shows the user a state X and the user says to change it to state Y, the thing we want to tell the server is to please change in the direction X → Y. We basically never want to translate that into something like "please flip between state X or Y, whichever one you currently have it in". (The only exception would be if the nature of "state X" and "state Y" is somehow such that that translation would be a natural interpretation of what the user actually intends.)

So, here, the UI we're showing either has the user as having already voted for the option, or as not having done so. If the UI says the user has already voted and the user hits the button, that means to remove their vote, unconditionally. Even if the server currently thinks they haven't voted, what the user asked for is to remove their vote, not add it — after all, they were looking at a UI that said they had voted.

One consequence of that general principle is that the canonical thing to do here wouldn't be to go look up `option.key` in the store again and find what it says. I think in this revision that will end up having the same behavior we want… but the canonical way to get that behavior is to look directly at the data structure that was used for rendering the UI. Here, `option.voters` is what controlled the list of names we showed. So how about we just inspect `option.voters` directly?

Reply:
> I think the real point there is that when the UI shows the user a state X and the user says to change it to state Y, the thing we want to tell the server is to please change in the direction X → Y. We basically never want to translate that into something like "please flip between state X or Y, whichever one you currently have it in". 

Yeah exactly that. The user couldn't instantly toggle the vote because it takes time to send an update to the server and then receive the event for it — that's not really a bad thing.

In the legacy mobile app, with locally echoing, you can have a sequence of states like this:
- User taps the button to unvote;
- User taps the button again to vote;
- The submessage event for the unvote arrives;
- The submessage event for the vote arrives;

which result in visually four toggles. This is probably because we don't have a way to deduplicate the updates. The current behavior of the this PR revision doesn't have that problem.